var/spack/repos/builtin/packages/py-warpx/package.py

```diff
@@ -38,4 +38,7 @@ class PyWarpx(PythonPackage):
     depends_on('warpx +lib +mpi +shared', type=('build', 'link'), when='+mpi')
 
     def setup_build_environment(self, env):
-        env.set('PYWARPX_LIB_DIR', self.spec['warpx'].prefix.lib)
+        if find(self.spec['warpx'].prefix.lib64, 'libwarpx.3d.so'):
+            env.set('PYWARPX_LIB_DIR', self.spec['warpx'].prefix.lib64)
+        else:
+            env.set('PYWARPX_LIB_DIR', self.spec['warpx'].prefix.lib)
```

Comment:
```suggestion
        env.set('PYWARPX_LIB_DIR', self.spec['warpx'].libs.directories[0])
```
We may need to create a `libs` attribute in the `warpx` package if the default search doesn't work.

Reply:
I made the above fix and got the following error:
Error: NoLibrariesError: Unable to recursively locate warpx libraries in ./spack/opt/spack/linux-rhel8-thunderx2/gcc-8.3.1/warpx-develop-ontb5kda42baggrxqff5nfmmcun2urr3

./spack/var/spack/repos/builtin/packages/py-warpx/package.py:41, in setup_build_environment:
         40    def setup_build_environment(self, env):
  >>     41        env.set('PYWARPX_LIB_DIR', self.spec['warpx'].libs.directories[0])


I was able to create and build the following symlinks.
lrwxrwxrwx 1 r1059 syssoft        14 Mar  8 11:00 libwarpx.so -> libwarpx.3d.so

I don't think it's detected due to a file suffix issue in the library name.

Reply:
You'll have to add a `libs` attribute to the `warpx` package so that it knows what name to search for. Try:
```python
@property
def libs(self):
    shared = '+shared' in self.spec
    return find_libraries(
        ['libwarpx.3d'], root=self.prefix, recursive=True, shared=shared
    )
```

Reply:
Thank you for suggesting a fix.

When I built it, I got the following error:
  File "setup.py", line 64, in run
    raise RuntimeError("CMake >= 3.15.0 is required")
RuntimeError: CMake >= 3.15.0 is required

I was able to build it by adding cmake's depend.

Reply:
Hm, if `py-warpx` starts to search for CMake it means it did not pick up the pre-compiled libraries.

In Spack, `warpx +lib` builds the library - and `py-warpx` just adds a lightweight Python layer.

Reply:
Why does the default search not work, though?